src/bin/common.rs

```diff
@@ -245,6 +257,43 @@ pub struct CliOptions {
   pub command: Option<Commands>,
 }
 
+fn get_version() -> &'static str {
+  static VERSION_STR: Lazy<String> = Lazy::new(|| {
+    format!(
+      "{} ({})",
+      built_info::GIT_VERSION
+        .map(|ver| if built_info::GIT_DIRTY.expect("git is true") {
+          format!("{}-MODIFIED", ver)
+        } else {
+          ver.to_string()
+        })
+        .unwrap_or_else(|| built_info::PKG_VERSION.to_string()),
+      // We cannot use `built_info::DEBUG` because that tells us if there are debug symbols,
+      // not if there are optimizations.
+      if cfg!(debug_assertions) { "Debug" } else { "Release" }
+    )
+  });
+  &VERSION_STR
+}
+
+fn get_long_version() -> &'static str {
+  static LONG_VERSION_STR: Lazy<String> = Lazy::new(|| {
+    format!(
+      "{}\n{} {}\nOptimizations: {}\nAssembly: {}\nThreading: {}\nUnstable Features: {}\nCompiler Flags: {}\nBuilt {}",
```

Comment:
Maybe it could also say "addition compiler flags"? If I just do `cargo build --release` it ends up with just an empty line after the "Compiler Flags:", which I think looks slightly strange. Maybe even something like `(none)` could be printed if there were no extra compiler flags, i.e. if the string was empty.

Reply:
Good call, I didn't even think to test that case 😅